examples/basic/main.tf

```diff
@@ -40,4 +40,5 @@ module "read_only_replica_postgresql_db" {
   remote_leader_crn = module.postgresql_db.crn
   member_memory_mb  = 2304  # Must be an increment of 384 megabytes. The minimum size of a read-only replica is 2 GB RAM
   member_disk_mb    = 15360 # Must be an increment of 512 megabytes. The minimum size of a read-only replica is 15.36 GB of disk
+  depends_on        = [module.postgresql_db]
```

Comment:
using explicit `depends_on` like this is not recommended by terraform unless there is no other option (see nice [article](https://itnext.io/beware-of-depends-on-for-modules-it-might-bite-you-da4741caac70) on this). In this case there is already an implicit dependency in place due to the fact this line: `remote_leader_crn = module.postgresql_db.crn` so no need to add an explicit one. 

For the problem you are trying to solve, please re-check the details I added to https://github.ibm.com/GoldenEye/issues/issues/5484 on what the solution had to be here. We need to either add a hard sleep, or call an api to check status before attempting to delete the read only replica. This is what the ICD team recommended in the support case I had opened